connector/src/main/java/org/geysermc/connector/utils/SkinUtils.java

```diff
@@ -60,25 +62,28 @@
 
         return buildEntryManually(
                 session,
-                profile.getId(),
-                profile.getName(),
-                geyserId,
+                playerEntity.getProfile().getId(),
+                playerEntity.getProfile().getName(),
+                playerEntity.getGeyserId(),
                 skin.getTextureUrl(),
                 skin.getSkinData(),
                 cape.getCapeId(),
                 cape.getCapeData(),
-                geometry.getGeometryName(),
-                geometry.getGeometryData()
+                geometry
         );
     }
 
     public static PlayerListPacket.Entry buildEntryManually(GeyserSession session, UUID uuid, String username, long geyserId,
                                                             String skinId, byte[] skinData,
                                                             String capeId, byte[] capeData,
-                                                            String geometryName, String geometryData) {
+                                                            SkinProvider.SkinGeometry geometry) {
+        if (geometry == SkinProvider.SKULL_GEOMETRY) {
+            // Prevents https://cdn.discordapp.com/attachments/613194828359925800/779458146191147008/unknown.png
+            skinId = skinId + "_skull";
+        }
```

Comment:
Even more reasons to make a method specific for Skulls

Reply:
I'm not opposed, if we can handle it without duplicating code.

Reply:
I think we need skinId to include the geometry hash as well.